File: src/main/java/net/auoeke/headsdowndisplay/config/HDDConfig.java
package net.auoeke.headsdowndisplay.config;

import me.sargunvohra.mcmods.autoconfig1u.ConfigData;
import me.sargunvohra.mcmods.autoconfig1u.annotation.Config;
import me.sargunvohra.mcmods.autoconfig1u.annotation.Config.Gui.Background;
import me.sargunvohra.mcmods.autoconfig1u.annotation.ConfigEntry.Category;
import me.sargunvohra.mcmods.autoconfig1u.annotation.ConfigEntry.Gui.Excluded;
import me.sargunvohra.mcmods.autoconfig1u.annotation.ConfigEntry.Gui.TransitiveObject;
import net.auoeke.headsdowndisplay.HDD;
import net.auoeke.headsdowndisplay.config.arm.ArmConfiguration;
import net.auoeke.headsdowndisplay.config.hotbar.Hotbar;

@Config(name = HDD.ID)
@Background("textures/block/andesite.png")
public class HDDConfig implements ConfigData {
    @Excluded
    public static transient HDDConfig instance;

    @TransitiveObject
    @Category("default")
    public Hotbar hotbar = new Hotbar();

    @TransitiveObject
    @Category("arm")
    public ArmConfiguration arm = new ArmConfiguration();

    @Override
    public void validatePostLoad() {
        this.hotbar.fadeEnd = this.hotbar.fadeDelay + this.hotbar.fadeDuration;
    }
}

File: source/headdowndisplay/config/HDDConfig.kt
package headdowndisplay.config

import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config
import headdowndisplay.HDD
import me.shedaniel.autoconfig.annotation.Config.Gui.Background
import me.shedaniel.autoconfig.annotation.ConfigEntry.Category
import me.shedaniel.autoconfig.annotation.ConfigEntry.Gui.TransitiveObject
import headdowndisplay.config.hotbar.Hotbar
import headdowndisplay.config.arm.ArmConfiguration

@Config(name = HDD.ID)
@Background("textures/block/andesite.png")
class HDDConfig : ConfigData {
    @JvmField
    @TransitiveObject
    @Category("default")
    var hotbar = Hotbar()

    @JvmField
    @TransitiveObject
    @Category("arm")
    var arm = ArmConfiguration()

    override fun validatePostLoad() {
        hotbar.fadeEnd = hotbar.fadeDelay + hotbar.fadeDuration
    }

    companion object {
        @Transient
        lateinit var instance: HDDConfig
    }
}
